smt.go

```diff
@@ -340,11 +344,15 @@ func (smt *SparseMerkleTree) sideNodesForRoot(path []byte, root []byte) ([][]byt
 		leftNode, rightNode := smt.th.parseNode(currentData)
 
 		// Get sidenode depending on whether the path bit is on or off.
-		if hasBit(path, i) == right {
-			sideNodes[i] = leftNode
+		if getBitAtFromMSB(path, i) == right {
+			sideNodes = append(sideNodes, nil)
+			copy(sideNodes[1:], sideNodes)
+			sideNodes[0] = leftNode
 			nodeHash = rightNode
 		} else {
-			sideNodes[i] = rightNode
+			sideNodes = append(sideNodes, nil)
```

Comment:
Wouldn't it be cleaner to add data to slice as before, and then explicitly reverse slice after the loop? (with a short comment why)

Reply:
Not really? That would both obfuscate the true intention to a later part of the code without an obvious link to the reader, and it would require a bunch of copying, reducing performance. Prepending is _self-documenting_ (/s but also kind of not).

Reply:
Fixed in 2dad41f.